agent/consul/fsm/commands_oss_test.go

```diff
@@ -1355,3 +1356,52 @@ func TestFSM_CABuiltinProvider(t *testing.T) {
 		assert.Equal(expected, state)
 	}
 }
+
+func TestFSM_ConfigEntry(t *testing.T) {
+	t.Parallel()
+
+	require := require.New(t)
+	fsm, err := New(nil, os.Stderr)
+	require.NoError(err)
+
+	// Create a simple config entry
+	entry := &structs.ProxyConfigEntry{
+		Kind: structs.ProxyDefaults,
+		Name: "global",
+		Config: map[string]interface{}{
+			"DestinationServiceName": "foo",
```

Comment:
nit: this is fine since it's a bag of stuff but it's probably confusing for future us that you chose the name of a field in a related-but-not-the-same part of the proxy config in this test case. This `Config` will be merged into the `Proxy.Config` while this field mirrors `Proxy.DestinationServiceName`. It's minor but maybe just to avoid confusion make this be `"foo": "bar"` or something?